service.go

```diff
@@ -30,31 +29,30 @@ const (
 	host = "host"
 )
 
-// Component interface for implementing service components.
+// Component interface for implementing Service components.
 type Component interface {
 	Run(ctx context.Context) error
 }
 
-// service is responsible for managing and setting up everything.
-// The service will start by default an HTTP component in order to host management endpoint.
-type service struct {
+// Service is responsible for managing and setting up everything.
+// The Service will start by default an HTTP component in order to host management endpoint.
+type Service struct {
 	name              string
+	version           string
 	cps               []Component
-	routesBuilder     *patronhttp.RoutesBuilder
-	middlewares       []middleware.Func
-	acf               patronhttp.AliveCheckFunc
-	rcf               patronhttp.ReadyCheckFunc
 	termSig           chan os.Signal
 	sighupHandler     func()
 	uncompressedPaths []string
 	httpRouter        http.Handler
+	errors            []error
```

Comment:
We may not need this as a field of Service? Given that it is only used in the construction/instantiation of the service, it can be a local scoped slice